Tinyxml2Serializer.cpp

```diff
@@ -3651,130 +3647,133 @@ std::string CTinyxml2Serializer::Set6DOFBodySettings(const std::vector<SSettings
-    //                    }
-    //                }
-
-    //                xml.AddElem("Markers");
-    //                xml.IntoElem();
-    //                {
-    //                    for (const auto& marker : segment.markers)
-    //                    {
-    //                        xml.AddElem("Marker");
-    //                        xml.AddAttrib("Name", marker.name.c_str());
-    //                        xml.IntoElem();
-    //                        {
-    //                            xml.AddElem("Position");
-    //                            xml.AddAttrib("X", std::to_string(marker.position.x).c_str());
-    //                            xml.AddAttrib("Y", std::to_string(marker.position.y).c_str());
-    //                            xml.AddAttrib("Z", std::to_string(marker.position.z).c_str());
-    //                            xml.AddElem("Weight", std::to_string(marker.weight).c_str());
-    //                        }
-    //                        xml.OutOfElem(); // Marker
-    //                    }
-    //                }
-    //                xml.OutOfElem(); // MarkerS
-
-    //                xml.AddElem("RigidBodies");
-    //                xml.IntoElem();
-    //                {
-    //                    for (const auto& rigidBody : segment.bodies)
-    //                    {
-    //                        xml.AddElem("RigidBody");
-    //                        xml.AddAttrib("Name", rigidBody.name.c_str());
-    //                        xml.IntoElem();
-
-    //                        xml.AddElem("Transform");
-    //                        xml.IntoElem();
-
-    //                        xml.AddElem("Position");
-    //                        xml.AddAttrib("X", std::to_string(rigidBody.position.x).c_str());
-    //                        xml.AddAttrib("Y", std::to_string(rigidBody.position.y).c_str());
-    //                        xml.AddAttrib("Z", std::to_string(rigidBody.position.z).c_str());
-    //                        xml.AddElem("Rotation");
-    //                        xml.AddAttrib("X", std::to_string(rigidBody.rotation.x).c_str());
-    //                        xml.AddAttrib("Y", std::to_string(rigidBody.rotation.y).c_str());
-    //                        xml.AddAttrib("Z", std::to_string(rigidBody.rotation.z).c_str());
-    //                        xml.AddAttrib("W", std::to_string(rigidBody.rotation.w).c_str());
-    //                        xml.OutOfElem(); // Transform
-
-    //                        xml.AddElem("Weight", std::to_string(rigidBody.weight).c_str());
-    //                        xml.OutOfElem(); // RigidBody
-    //                    }
-    //                }
-    //                xml.OutOfElem(); // RigidBodies
-    //            }
-    //            xml.OutOfElem(); // Segment
-
-    //            for (const auto& childSegment : segment.segments)
-    //            {
-    //                xml.IntoElem();
```

Comment:
Question: What was the significance of these in the loop? They are removed now

Reply:
The code in the loop was required for CMarkup, since CMarkup needs to manually traverse the hierarchy (_**IntoElem()** and **OutOfElem()**_). In tinyxml2, the function directly places new elements into their parent (**InsertEndChild()**). Compared to the previous code, the nested is preserved since each **recurseSegments()** call gets the appropriate parent element as an argument.